core/src/test/java/com/datastax/oss/driver/internal/core/pool/ChannelPoolReconnectTest.java

```diff
@@ -133,4 +136,59 @@ public void should_reconnect_when_channel_starts_graceful_shutdown() throws Exce
 
     factoryHelper.verifyNoMoreCalls();
   }
+
+  @Test
+  public void should_let_current_attempt_complete_when_reconnecting_now()
```

Comment:
I added a debug log to JAVA-1568.  The test where this happens is `HeartbeatIT#should_close_connection_when_heartbeat_times_out()`.  Although it's very hard to reproduce in isolation, but I see it frequently when running the entire short suite.